repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -523,7 +521,14 @@ public Map<AtlasVertex, List<AtlasVertex>> getClassificationPropagatedEntitiesMa
                 String            classificationId      = classificationVertex.getIdForDisplay();
                 String            sourceEntityId        = getClassificationEntityGuid(classificationVertex);
                 AtlasVertex       sourceEntityVertex    = AtlasGraphUtilsV2.findByGuid(this.graph, sourceEntityId);
-                List<AtlasVertex> entitiesPropagatingTo = getImpactedVerticesV2(sourceEntityVertex, relationshipGuidToExclude, classificationId);
+                String propagationMode = CLASSIFICATION_PROPAGATION_MODE_DEFAULT;
+
+                if(toAtlasClassification(classificationVertex).getRestrictPropagationThroughLineage()){
```

Comment:
instead of converting to AtlasClassification, fetch just required property value from the vertex
converting to AtlasClassification will read all properties from vertex where as we just need single property value